src/ttt/core.clj

```diff
@@ -1,51 +1,31 @@
 (ns ttt.core
-  (:require [clojure.math.numeric-tower :as math]))
-
-;====BOARD====
-(defn initial-board 
-  ([board-size]
-   (into [] (range (* board-size board-size))))
-   ([]
-    (initial-board 3)))
-
-(defn add-padding
-  [row]
-  (clojure.string/join " | " row))
-
-(defn print-board [board]
-  (let [size (math/sqrt (count board))]
-    (let [rows (into [] (map add-padding (partition size board)))
-          divider (str "\n" (clojure.string/join (repeat (count (first rows)) "-")) "\n")]
-      (println (str "\n" (clojure.string/join divider rows))))))
-
-
-;====Player=====
-(defn player 
-  [marker]
-  {:marker marker})
-
-(def players [(player "X") (player "O")])
-
-(defn current-player
-  [[player1 player2]]
-  player1)
-
-;====UI=====
-(defn start-game-message
+  (:gen-class)
+  (:require [ttt.board :as board]
+            [ttt.player :as player]
+            [ttt.ui :as ui]))
+
+(def game-state (atom {:board (board/create-initial-board)
+                       :players player/players
+                       :current-player (first player/players)}))
+
+(defn set-game-state
+  [board players]
+  (swap! game-state 
+        (fn [current-state]
+           (merge current-state {:board board :players players :current-player (first players)}))))
+
+(defn play 
   []
-  (println "Let's play tic-tac-toe.")
-  (first-player-message players)
-  (println "Enter a number between 0-8 to select a spot."))
+  (ui/print-board (:board @game-state))
```

Comment:
I'll have to read up on what `@game-state` means.  Is this some sort of global variable?

If you need a way to maintain a game state as it changes with one turn to another, you can always transform references/mutations of a global state to passing the state in/out of the functions that change it.

So instead of mutating state (usually highlighted in red and identified with `!` in the name)

```scheme
(define game-state (make-game-state ...))

(define (take-turn)
  (set! game-state (player/pick-move)))
```

you can transform this to a pure functional structure

```scheme
(let loop ([game-state (create-game)])
  (if (game-over? game-state)
      (game-winner game-state)
      (loop (player/pick-move game-state))))
```

Reply:
The big idea is to pass `game-state` into the functional that calculates and returns the next game state.  Just keep passing it in and returning it from whatever functions need it.

Reply:
And if things weren't already confusing enough, those examples are in Scheme instead of Clojure.

Reply:
the @game-state is a clojure atom! It's a way to keep related data together. when I need to update the data, i call `swap!` and it will merge the old data with the new data and output a new state with the same reference name. I learned about it here https://www.braveclojure.com/zombie-metaphysics/. I'm going to be passing @game-state around to update after every turn.